src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -484,4 +489,76 @@ else if (tokenType == TokenTypes.LITERAL_PRIVATE) {
         return illegalClassNames;
     }
 
+    /**
+     * Strip initial newline and preceding whitespace on each line from text block content.
+     * In order to be consistent with how javac handles this task, we have modeled this
+     * implementation after the code from:
+     * github.com/openjdk/jdk14u/blob/master/src/java.base/share/classes/java/lang/String.java
+     *
+     * @param textBlockContent the actual content of the text block.
+     * @return string consistent with javac representation.
+     */
+    public static String stripIndentAndInitialNewLineFromTextBlock(String textBlockContent) {
+        final String contentWithInitialNewLineRemoved =
+            ALL_NEW_LINES.matcher(textBlockContent).replaceFirst("");
+        final List<String> lines =
+            Arrays.asList(ALL_NEW_LINES.split(contentWithInitialNewLineRemoved));
+        final int indent = getSmallestIndent(lines);
+        final String suffix = "";
+
+        return lines.stream()
+                .map(line -> stripIndentFromLine(line, indent))
+                .collect(Collectors.joining(System.lineSeparator(), "", suffix));
+    }
+
+    /**
+     * Helper method for stripIndentAndInitialNewLineFromTextBlock, strips correct indent
+     * from string, or returns empty string if no text.
+     *
+     * @param line the string to strip indent from
+     * @param indent the amount of indent to remove
+     * @return modified string with removed indent, or empty string.
+     */
+    private static String stripIndentFromLine(String line, int indent) {
+        final int firstNonWhitespace = indexOfNonWhitespace(line);
+        final int lastNonWhitespace = line.length();
```

Comment:
nit: is this name accurate? (can the string end in whitespace?)

Reply:
You have brought up a good point:

```
➜  text-blocks-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.

➜  text-blocks-grammar cat TestClass.java
public class TestClass {
public static void main (String... args) {
    String str9a = """
            test    """;
    String str9b = "test    ";

    if (str9a.equals(str9b)) {
        System.out.println("str9a equals str9b");
    }
    else {
        System.out.println("str9a does not equal str9b");
    }
    System.out.println(str9a);
    System.out.println(str9b);
}
}

➜  text-blocks-grammar /home/nick/.jdks/openjdk-14.0.1/bin/java --enable-preview -classpath /home/nick/Desktop/text-blocks-grammar/out/production/text-blocks-grammar TestClass 
str9a does not equal str9b
test
test     
```
Looks like we need to change this behavior.

Reply:
Behavior and name is updated.